### ISSUE TITLE
Local repositories contained in a Package are not found by Resolver

### ISSUE THREAD

**Issue by bhallionOhbibi** [googlesamples/unity-jar-resolver#373]
### Please fill in the following fields:
Unity editor version: **2019.2.14**
External Dependency Manager version: **1.2.153**
Source you installed EDM (from .unitypackage or Unity Package Manager): **.unitypackage**
Features in External Dependency Manager in use (Android Resolver, iOS Resolver, VersionHandler, etc.): **Android Resolver**
Plugins SDK in use (Firebase, Admob, Facebook, etc.): 
Platform you are using the Unity editor on (Mac, Windows, or Linux): **Windows**
Platform you are targeting (iOS, Android, and/or desktop): **Android**
Scripting Runtime (Mono, and/or IL2CPP): **Mono**

### Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)

I have a local package which contains both a PluginDependencies.xml and a m2repository.

> 
> MyProject/Assets/ExternalDependencyManager
> MyProject/CustomPackages/myplugin/m2repository                <--
> MyProject/CustomPackages/myplugin/AppDepencencies.xml    <--

This local package is added to unity using Packages/manifest.json. Its content is available as any other Packages inside my Unity Editor.

The PluginDependencies.xml declare an androidPackage whom repository is local to the package like so: 
```
    <androidPackage spec="com.company.plugin:company-plugin:1.0.1">
      <repositories>
        <repository>CustomPackages/myplugin/m2repository</repository>
      </repositories>
    </androidPackage>
```

Here are my AndroidResolver Settings:
```
<projectSettings>
  <projectSetting name="com.google.external-dependency-managerAnalyticsEnabled" value="False" />
  <projectSetting name="Google.IOSResolver.AutoPodToolInstallInEditor" value="True" />
  <projectSetting name="Google.IOSResolver.CocoapodsIntegrationMethod" value="2" />
  <projectSetting name="Google.IOSResolver.PodfileEnabled" value="True" />
  <projectSetting name="Google.IOSResolver.PodToolExecutionViaShellEnabled" value="True" />
  <projectSetting name="Google.IOSResolver.VerboseLoggingEnabled" value="False" />
  <projectSetting name="Google.PackageManagerResolver.Enable" value="False" />
  <projectSetting name="Google.PackageManagerResolver.PromptToAddRegistries" value="True" />
  <projectSetting name="Google.PackageManagerResolver.PromptToMigratePackages" value="True" />
  <projectSetting name="Google.PackageManagerResolver.VerboseLoggingEnabled" value="False" />
  <projectSetting name="Google.UnityPackageManagerResolver.Enable" value="False" />
  <projectSetting name="Google.UnityPackageManagerResolver.VerboseLoggingEnabled" value="False" />
  <projectSetting name="Google.VersionHandler.VerboseLoggingEnabled" value="False" />
  <projectSetting name="GooglePlayServices.AndroidPackageInstallationEnabled" value="True" />
  <projectSetting name="GooglePlayServices.AutoResolutionDisabledWarning" value="False" />
  <projectSetting name="GooglePlayServices.AutoResolveOnBuild" value="False" />
  <projectSetting name="GooglePlayServices.AutoResolverEnabled" value="False" />
  <projectSetting name="GooglePlayServices.ExplodeAars" value="False" />
  <projectSetting name="GooglePlayServices.LocalMavenRepoDir" value="Assets/GeneratedLocalRepo" />
  <projectSetting name="GooglePlayServices.PatchAndroidManifest" value="True" />
  <projectSetting name="GooglePlayServices.PatchMainTemplateGradle" value="True" />
  <projectSetting name="GooglePlayServices.PromptBeforeAutoResolution" value="False" />
  <projectSetting name="GooglePlayServices.UseGradleDaemon" value="False" />
  <projectSetting name="GooglePlayServices.UseJetifier" value="True" />
  <projectSetting name="GooglePlayServices.VerboseLogging" value="False" />
</projectSettings>
```

And the error I get: 
```

CustomPackages/myplugin/AppDepencencies.xml:20: Repo path 'CustomPackages/myplugin/GeneratedLocalRepo/m2repository' does not exist.
UnityEngine.Debug:LogWarning(Object)
Google.Logger:Log(String, LogLevel) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/Logger.cs:129)
GooglePlayServices.PlayServicesResolver:Log(String, LogLevel) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:989)
GooglePlayServices.GradleResolver:RepoPathToUri(String, String) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/GradleResolver.cs:228)
GooglePlayServices.GradleResolver:DependenciesToRepoUris(IEnumerable`1) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/GradleResolver.cs:271)
GooglePlayServices.PlayServicesResolver:GetRepos(IEnumerable`1) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1927)
GooglePlayServices.PlayServicesResolver:GradleMavenReposLines(ICollection`1) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1957)
GooglePlayServices.GradleTemplateResolver:InjectDependencies(ICollection`1) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/GradleTemplateResolver.cs:326)
GooglePlayServices.<ResolveUnsafe>c__AnonStorey20:<>m__40() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1807)
Google.RunOnMainThread:ExecuteNext() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:486)
Google.RunOnMainThread:<ExecuteAllUnnested>m__B() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:536)
Google.RunOnMainThread:RunAction(Action) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:343)
Google.RunOnMainThread:ExecuteAllUnnested(Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:530)
Google.RunOnMainThread:Run(Action, Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:472)
GooglePlayServices.PlayServicesResolver:ResolveUnsafe(Action`1, Boolean, Boolean, Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1806)
GooglePlayServices.<ScheduleResolve>c__AnonStorey1E:<>m__3E() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1637)
GooglePlayServices.PlayServicesResolver:ExecuteNextResolveJob() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1494)
GooglePlayServices.PlayServicesResolver:ScheduleResolve(Boolean, Boolean, Action`1, Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1650)
GooglePlayServices.PlayServicesResolver:ExecuteMenuResolve(Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1861)
GooglePlayServices.PlayServicesResolver:MenuForceResolve() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1885)
```

Also I tried to use a absolute path like this:
```
    <androidPackage spec="com.company.plugin:company-plugin:1.0.1">
      <repositories>
        <repository>F:/DEV/MyProject/CustomPackages/myplugin/m2repository</repository>
      </repositories>
    </androidPackage>
```
 and I got a different error:

```
Job failed with exception: System.ArgumentException: An empty file name is not valid.
  at System.IO.DirectoryInfo.CheckPath (System.String path) [0x00016] in <567df3e0919241ba98db88bec4c6696f>:0 
  at System.IO.DirectoryInfo..ctor (System.String path, System.Boolean simpleOriginalPath) [0x00006] in <567df3e0919241ba98db88bec4c6696f>:0 
  at System.IO.DirectoryInfo..ctor (System.String path) [0x00000] in <567df3e0919241ba98db88bec4c6696f>:0 
  at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo..ctor(string)
  at Google.FileUtils.CreateFolder (System.String path) [0x0000d] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:630 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at Google.FileUtils.CreateFolder (System.String path) [0x0001b] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:632 
  at GooglePlayServices.PlayServicesResolver.CopyAssetAndLabel (System.String sourceLocation, System.String targetLocation, System.Boolean force) [0x00075] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:2230 
  at GooglePlayServices.GradleTemplateResolver.CopySrcAars (System.Collections.Generic.ICollection`1[T] dependencies) [0x00082] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/GradleTemplateResolver.cs:112 
  at GooglePlayServices.GradleTemplateResolver.InjectDependencies (System.Collections.Generic.ICollection`1[T] dependencies) [0x00134] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/GradleTemplateResolver.cs:307 
  at GooglePlayServices.PlayServicesResolver+<ResolveUnsafe>c__AnonStorey20.<>m__40 () [0x00000] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1807 
  at Google.RunOnMainThread.ExecuteNext () [0x0003d] in /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:486 
UnityEngine.Debug:LogError(Object)
Google.RunOnMainThread:ExecuteNext() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:488)
Google.RunOnMainThread:<ExecuteAllUnnested>m__B() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:536)
Google.RunOnMainThread:RunAction(Action) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:343)
Google.RunOnMainThread:ExecuteAllUnnested(Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:530)
Google.RunOnMainThread:Run(Action, Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:472)
GooglePlayServices.PlayServicesResolver:ResolveUnsafe(Action`1, Boolean, Boolean, Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1806)
GooglePlayServices.<ScheduleResolve>c__AnonStorey1E:<>m__3E() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1637)
GooglePlayServices.PlayServicesResolver:ExecuteNextResolveJob() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1494)
GooglePlayServices.PlayServicesResolver:ScheduleResolve(Boolean, Boolean, Action`1, Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1650)
GooglePlayServices.PlayServicesResolver:ExecuteMenuResolve(Boolean) (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1861)
GooglePlayServices.PlayServicesResolver:MenuForceResolve() (at /Users/chkuang/Workspace/Git/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:1885)
```

### Please answer the following, if applicable:
What's the issue repro rate? **100%**

### Remarks: 
Note that If I move my m2repository into the 'Assets' folder, everything work fine.

Also note that while this example was made generic, I'm attempting to create package version of existing plugins which are using EDM. In that case: Firebase

Having a plugin implementation/structure that support both being included in Assets/ and as a package in the new unity package system is the future and already offers many advantages.
I hope resolving this will help achieve this goal.

**Comment by bhallionOhbibi** [comment]
Hi Shawn

Thank you for taking the time to adress my issue.

More details about my issue.

The main ingredients is:
- Declare a repository, with a path relative to your project root, outside of _Assets/_ or _Packages/_ ([example](https://github.com/bhallionOhbibi/EDMWithCustomPackage/blob/533f97e307c3efba3a713d1c46b1331d358d0d67/CustomPackages/Packages/firebasesdk/Firebase/Editor/AppDependencies.xml#L18) <- declared in _CustomPackages/_)

I created a repo to help you reproduce the issue: https://github.com/bhallionOhbibi/EDMWithCustomPackage

**How to reproduce:**
1. Open the project with unity 2019 (I did my test with Unity 2019.3.8f1)
2. Assets > External Dependency Manager > Android Resolver > Force Resolve
3. Observe exception thrown by:
https://github.com/googlesamples/unity-jar-resolver/blob/76c3955ea21a8afe2deed02f1f84f27764a1c35c/source/VersionHandlerImpl/src/FileUtils.cs#L627

**Also**
I gave it a look and I think the issue comes from this:
https://github.com/googlesamples/unity-jar-resolver/blob/76c3955ea21a8afe2deed02f1f84f27764a1c35c/source/AndroidResolver/src/GradleTemplateResolver.cs#L99

**Finaly**

I also tried to move the package location from _CustomPackages/Packages/_ to _Packages/Packages/_

And I had an issue with this method which seems to have trouble moving files from a custom package location to the asset folder: 
https://github.com/googlesamples/unity-jar-resolver/blob/76c3955ea21a8afe2deed02f1f84f27764a1c35c/source/AndroidResolver/src/PlayServicesResolver.cs#L2281

**Comment by chkuang-g** [comment]
Sorry for the late reply.

I took a look of your repo and I noticed that you include the entire Firebase Unity SDK under `CustomPackages/Packages/firebasesdk` folder.  I do not think Firebase SDK would work at all in the way.  It needs to be under `Assets` folder or `Library/PackageCache` if it is installed through UPM.  A lot of functionality will simply break if you do so.

If you really like to keep your project clean, I would recommend you to install Firebase through Unity Package Manager [instruction here](https://firebase.google.com/docs/unity/setup#add-sdks-upm)

Hope this helps.
Shawn
